src/python/pants/cache/local_artifact_cache.py

```diff
@@ -129,28 +142,15 @@ def prune(self, root):
       safe_rm_oldest_items_in_dir(root, max_entries_per_target)
 
   def has(self, cache_key):
-    return self._artifact_for(cache_key).exists()
-
-  def _artifact_for(self, cache_key):
-    return self._artifact(self._cache_file_for_key(cache_key))
+    return self._artifact(self._cache_file_for_key(cache_key)).exists()
 
   def use_cached_files(self, cache_key, results_dir=None):
-    tarfile = self._cache_file_for_key(cache_key)
+    tarball = self._cache_file_for_key(cache_key)
     try:
-      artifact = self._artifact_for(cache_key)
-      if artifact.exists():
-        if results_dir is not None:
-          safe_rmtree(results_dir)
-        artifact.extract()
-        return True
-    except Exception as e:
-      # TODO(davidt): Consider being more granular in what is caught.
-      logger.warn('Error while reading {0} from local artifact cache: {1}'.format(tarfile, e))
-      safe_delete(tarfile)
```

Comment:
Continuing to delete artifacts that we fail to extract would seem like a good idea... it's unclear how often we hit this codepath in practice, but it's possible it's been self-healing and avoiding file corruption in some cases.